### ISSUE TITLE
Nested annotations aren't popped

### ISSUE THREAD

**Issue by Alxandr** [Marwes/pretty.rs#56]
If you nest 2 annotations directly inside each other, it doesn't seem to pop them. I wrote some code to log what's actually being written to my writer, and this is the result:

```
Input: Annotated(Span(339570348708772992), Annotated(Span(13962826062594635996), Annotated(Style(Style { color: Some(Green), bold: false }), Text("b\'b\'"))))
Log: ["push: Span(339570348708772992)", "push: Span(13962826062594635996)", "push: Style(Style { color: Some(Green), bold: false })", "write: b\'b\'", "pop annotation"]
```

As you can see, even though there are 3 annotation pushes, there is only one pop.

**Comment by Alxandr** [comment]
It seems to be fixed by replacing the `if` [here](https://github.com/Marwes/pretty.rs/blob/master/src/render.rs#L383) with a `while`.

**Comment by Marwes** [comment]
Yep, just finished a fix https://github.com/Marwes/pretty.rs/pull/57 ! Thanks for reporting, this broke while I was optimizing the inner loop here to avoid `Vec::push/Vec::pop`

**Comment by Marwes** [comment]
Released as 0.7.1

**Comment by Alxandr** [comment]
Thanks :)
